src/api.ts

```diff
@@ -74,21 +74,28 @@ export function putForm(data: {
 export function downloadFile(
   url: string,
   key: string,
-  version: string,
-): Promise<String> {
+  server_update_time: Date,
+): any {
   const path = `${DIRS.DocumentDir}/${key}`;
 
-  return AsyncStorage.getItem(key).then((val: any) =>
-    val === version
-      ? path
-      : RNFetchBlob.config({
+  return AsyncStorage.getItem(key).then((cache_update_time : string|null) => {
+      //Transform string taken from storage to date object so it can be compared
+      if (cache_update_time !== null){
+        const cache_date_object = new Date(cache_update_time)
+
+        if (cache_date_object.getTime() == server_update_time.getTime()){
+          return path
+        }
+      }
+      RNFetchBlob.config({
```

Comment:
Should be returning the Promise, as otherwise `downloadFile` returns `Promise<void>`.